examples/java-matter-controller/java/src/com/matter/controller/commands/common/Command.java

```diff
@@ -0,0 +1,243 @@
+/*
+ *   Copyright (c) 2022 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+package com.matter.controller.commands.common;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+public abstract class Command {
+  private String mName = null;
+  private String mHelpText = null;
+
+  private boolean mIsInteractive = false;
+  private ArrayList<Argument> mArgs = new ArrayList<Argument>();
+
+  public Command(String commandName) {
+    this.mName = commandName;
+  }
+
+  public Command(String commandName, String helpText) {
+    this.mName = commandName;
+    this.mHelpText = helpText;
+  }
+
+  public final String getName() {
+    return mName;
+  }
+
+  public final String getHelpText() {
+    return mHelpText;
+  }
+
+  public final String getAttribute() {
```

Comment:
annotate with `@Nullable` or use exceptions (probably using exceptions is better).

Then again what does a `getAttribute` even mean? Why is there a single value that is expected to match here? This is odd .... it should probably be `getFirstOfType()` and pass in Attribute.